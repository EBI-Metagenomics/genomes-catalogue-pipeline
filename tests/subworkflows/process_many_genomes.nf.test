nextflow_workflow {

    name "Test Workflow PROCESS_MANY_GENOMES"
    script "subworkflows/process_many_genomes.nf"
    workflow "PROCESS_MANY_GENOMES"

    test("Should run without failures") {

        when {

            workflow {
                """
                input[0] = channel.from([
                    tuple(
                        "CAJLGA01",
                        file("$baseDir/tests/subworkflows/fixtures/process_many_genomes/cluster_CAJLGA01/CAJLGA01/CAJJTO01.fa")
                    ),
                    tuple(
                        "CAJLGA01",
                        file("$baseDir/tests/subworkflows/fixtures/process_many_genomes/cluster_CAJLGA01/CAJLGA01/CAJKGB01.fa")
                    ),
                    tuple(
                        "CAJLGA01",
                        file("$baseDir/tests/subworkflows/fixtures/process_many_genomes/cluster_CAJLGA01/CAJLGA01/CAJLGA01.fa")
                    ),
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.out.prokka_faas,
                workflow.out.prokka_fnas,
                workflow.out.prokka_gffs,
                workflow.out.rep_prokka_fna,
                workflow.out.rep_prokka_gff,
                workflow.out.rep_prokka_faa,
                workflow.out.rep_prokka_gbk,
                workflow.out.rep_prokka_ffn,
                workflow.out.non_rep_prokka_fna,
                workflow.out.non_rep_prokka_gff,
                workflow.out.core_genes
            ).match()
        }
    }
}
