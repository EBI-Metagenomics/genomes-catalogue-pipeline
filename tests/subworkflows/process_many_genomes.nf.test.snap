{
    "Should run without failures": {
        "content": [
            [
                [
                    "CAJLGA01",
                    "CAJJTO01.faa:md5,87da0675d312c4089db76ff1ce534bdf"
                ],
                [
                    "CAJLGA01",
                    "CAJKGB01.faa:md5,ad7fd9b962c3d6417aed98b94dd8a32b"
                ],
                [
                    "CAJLGA01",
                    "CAJLGA01.faa:md5,f8739821dedeb6b8724881ed72016661"
                ]
            ],
            [
                [
                    "CAJLGA01",
                    "CAJJTO01.fna:md5,c6cd5c3773e2c8536caf0882b6b52b8a"
                ],
                [
                    "CAJLGA01",
                    "CAJKGB01.fna:md5,47cccbb99536d4d308a13088bc67b1b5"
                ],
                [
                    "CAJLGA01",
                    "CAJLGA01.fna:md5,7adfd09cc0deb5248c0ccf785bf0a8cd"
                ]
            ],
            [
                [
                    "CAJLGA01",
                    "CAJJTO01.gff:md5,454880f5eb2a6fd91fcfbdd365259f50"
                ],
                [
                    "CAJLGA01",
                    "CAJKGB01.gff:md5,919a04e2e13913d68b3222e94831aeae"
                ],
                [
                    "CAJLGA01",
                    "CAJLGA01.gff:md5,d612867e1c0bf28780775c264b08c83b"
                ]
            ],
            [
                [
                    "CAJLGA01",
                    "CAJLGA01.fna:md5,7adfd09cc0deb5248c0ccf785bf0a8cd"
                ]
            ],
            [
                [
                    "CAJLGA01",
                    "CAJLGA01.gff:md5,d612867e1c0bf28780775c264b08c83b"
                ]
            ],
            [
                [
                    "CAJLGA01",
                    "CAJLGA01.faa:md5,f8739821dedeb6b8724881ed72016661"
                ]
            ],
            [
                [
                    "CAJLGA01",
                    "CAJLGA01.gbk:md5,1c3daf1527dab292e01f2cf86448f2c9"
                ]
            ],
            [
                [
                    "CAJLGA01",
                    "CAJLGA01.ffn:md5,a53fc09c41d9485378d3d4d1dc464ba0"
                ]
            ],
            [
                [
                    "CAJLGA01",
                    "CAJJTO01.fna:md5,c6cd5c3773e2c8536caf0882b6b52b8a"
                ],
                [
                    "CAJLGA01",
                    "CAJKGB01.fna:md5,47cccbb99536d4d308a13088bc67b1b5"
                ]
            ],
            [
                [
                    "CAJLGA01",
                    "CAJJTO01.gff:md5,454880f5eb2a6fd91fcfbdd365259f50"
                ],
                [
                    "CAJLGA01",
                    "CAJKGB01.gff:md5,919a04e2e13913d68b3222e94831aeae"
                ]
            ],
            [
                [
                    "CAJLGA01",
                    "CAJLGA01.core_genes.txt:md5,99b02869a691be679e37b4f908a11b63"
                ]
            ]
        ],
        "timestamp": "2023-04-25T15:18:08+0000"
    }
}
